cloudinit/sources/DataSourceMAAS.py

```diff
@@ -164,6 +166,24 @@ def check_instance_id(self, sys_cfg):
         return self.id_hash == get_id_from_ds_cfg(ncfg)
 
 
+class DataSourceMAASLocal(DataSourceMAAS):
+    network_config_sources = (
+        NetworkConfigSource.CMD_LINE,
+        NetworkConfigSource.SYSTEM_CFG,
+        NetworkConfigSource.DS,
+        NetworkConfigSource.INITRAMFS,
+    )
+
+    def _get_data(self):
+        if not KlibcNetworkConfigSource().is_applicable():
```

Comment:
While it's not high priority, it would be nice to get some unittest coverage here to assert that _get_data is triggered in DataSourceMAASLocal bcased on KlibcNetworkConfigSource.